### PR TITLE
Add oswitch formula

### PR DESCRIPTION
**Pull request by brainstorm**
From https://github.com/wurmlab/oswitch/issues/39, thanks @yeban.

### CODE REVIEW

**Comment by tseemann** [comment]
Not sure what this means!

```
==> brew style oswitch
WARNING:  You don't have /Users/travis/.gem/ruby/2.2.0/bin in your PATH,
	  gem executables will not run.
Successfully installed tins-1.6.0
Successfully installed ruby-progressbar-1.7.5
Successfully installed ast-2.1.0
Successfully installed parser-2.2.3.0
Successfully installed astrolabe-1.3.1
Successfully installed powerpack-0.1.1
Successfully installed rainbow-2.0.0
Successfully installed rubocop-0.35.1
8 gems installed
Error: The 'rubocop' gem is installed but couldn't find 'rubocop' in the PATH:
/Users/travis/.gem/ruby/2.0.0/bin:/Users/travis/bin:/Users/travis/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/travis/.rvm/bin:/usr/local/Library/ENV/scm:/usr/local/Library/ENV/scm
==> FAILED
```

**Comment by tseemann** [comment]
Here is some info about Gems in Brew:
https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Gems,-Eggs-and-Perl-Modules.md

**Comment by brainstorm** [comment]
Humm, strange, it installs cleanly locally :/ I'll have a look at those pointers, thanks Torsten!

**Comment by yeban** [comment]
oswitch doesn't even use / depend on those gems. They are coming from `brew audit homebrew/oswitch.rb --strict --online` command: I get the same error running that command locally.

Also, I ran `brew audit homebrew/oswitch.rb` locally and got:

```
oswitch:
 * Stable: SHA1 checksums are deprecated, please use SHA256
 * Don't use ruby as a dependency. We allow non-Homebrew ruby installations.

Error: 2 problems in 1 formula
```

**Comment by yeban** [comment]
I'm guessing it's the dependency on ruby that's causing the formula to fail:

During the tests, oswitch is installed first and then audited. This likely installs Ruby 2.2 in the container / vm. Accordingly `gem install` (run by audit command) installs them in `/Users/.../2.2.0/gems`. But `/Users/.../2.2.0/bin` (where it will place the executables) doesn't get a chance to be added to the PATH.

**Comment by yeban** [comment]
Woot! That fixed it (on 10.11 - I was following the build log). There are some rubocop complaints now.

**Comment by yeban** [comment]
@brainstorm How can I help with the audit warnings and errors?

**Comment by yeban** [comment]
@brainstorm I will try and send you a PR to address audit errors tomorrow.

**Comment by yeban** [comment]
@brainstorm Did you get a chance to look at my [PR](https://github.com/brainstorm/homebrew-science/pull/2)? It hopefully resolves the interpolation issue.

**Comment by brainstorm** [comment]
Thanks @yeban, merged, let's see if the build succeeds now :+1:

**Comment by brainstorm** [comment]
Seems that the Travis build will always fail for github API errors... @homebrew-science, any way to have this fixed with some agreement with @github, perhaps?

Otoh, @yeban, seems that `gmt`, a sci library is the one erroring now, perhaps setting it as a `require gmt` requirement in the formula... but does it require gmt really?

**Comment by yeban** [comment]
oswitch only uses colorise gem. Rest is stdlib.

**Comment by brainstorm** [comment]
Yeah, it seems that the build status for Travis is now based on how lucky we get against the GitHub API rate limit. Since other formulas succeed without the rate limit errors, I'll just give it some more time before trying again, right @tseemann?

**Comment by tseemann** [comment]
Are you able to squash all the commits into a single commit?
eg. git rebase -i HEAD~10 and put "s" in front of all the extra ones etc.

**Comment by brainstorm** [comment]
@tseemann Just did so... had to `push -f` since stuff was already pushed, I hope I did not break anything :_S

**Comment by yeban** [comment]
Doesn't Travis automatically requeue builds that failed due to Github API failure?

File: oswitch.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Oswitch < Formula
+  desc "Docker environments to work with Bioinformatics tools"
+  homepage "https://github.com/yeban/oswitch"
+
```
Comment (dpo) [inline oswitch.rb]: No need for this blank line.

File: oswitch.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  url "https://github.com/yeban/oswitch/archive/v0.2.6.tar.gz"
+  sha256 "af3ce13305be8747242dc11bdd35360b568ea738be8c490d448915d4c675871f"
+
+  bottle do
```
Comment (dpo) [inline oswitch.rb]: You wouldn't have a `bottle` section in a new formula. The bot will add it automatically.

File: oswitch.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    # Re-write RubyGem generated bin stub to load oswitch from prefix.
+    inreplace "#{bin}/oswitch" do |s|
+      s.gsub!(/require 'rubygems'/,
+        "ENV['GEM_HOME']='#{prefix}'\nrequire 'rubygems'")
+    end
+  end
+
+  test do
+    print "No tests for this package"
```
Comment (dpo) [inline oswitch.rb]: Remove the `test` method if there's no useful test to run.

**Comment by yeban** [comment]
Thanks a lot @brainstorm. Thanks all. :)

**Pull request closed without merging**